### DISCUSSION TITLE
What's the difference between MaterixOne and SingleStore?

### DISCUSSION THREAD

**Discussion by Sunt-ing** [matrixorigin/matrixone#2234] [Q&A] [answered]
Hi, developers. I'm wondering what's the difference between MaterixOne and SingleStore?
Thanks!

**Comment by yingfeng** [comment, marked as answer]
Actually they have similar vision: single store,  but different approaches as well as scopes:
1. SS chooses a typical delta-main approach to provide htap, the delta is based on an in-memory row store and the main is columnar store, additionally the main area does not provide capabilities for frequent updating. While MO TAE is a pure columnar store based solution although it has mem-table, it's not a delta-main architecture.  
2. MO has the infrastructure to contain multiple storage engines, such as AOE(append only), TPE(kv engine), TAE(transactional analytical),  although it might be reduced to fewer or even one in future. Such infrastructure provides the possibility to integrate heterogeneous engines into single one.
3. MO has the builtin streaming scope which has not been started yet which has not been contained by SS

**Reply by Sunt-ing** [reply]
Thanks for your vivid and clear explanation!
